src/states/node.rs

```diff
@@ -809,6 +829,150 @@ impl Node {
+                }
+            }
+        }
+        result.map(Ok)
+    }
+
+    fn handle_node_approval(&mut self,
+                            groups: Vec<(Prefix<XorName>, Vec<PublicId>)>)
+                            -> Evented<Result<(), RoutingError>> {
+        let mut events = Evented::empty();
+        if self.is_approved {
+            warn!("{:?} Received duplicate NodeApproval.", self);
+            return events.with_value(Ok(()));
+        }
+
+        self.peer_mgr.populate_routing_table(&groups);
+
+        // TODO: is this neccssary as this node is not approved as a full node by the section yet
+        let our_prefix = *self.peer_mgr.routing_table().our_prefix();
+        self.send_section_list_signature(our_prefix, None);
+
+        let name = *self.name();
+        if let Err(error) = self.send_routing_message(RoutingMessage {
+                src: Authority::ManagedNode(name),
+                dst: Authority::Section(name),
+                content: MessageContent::ApprovalConfirmation,
+            })
+            .extract(&mut events) {
+            debug!("{:?} Failed sending ApprovalConfirmation: {:?}", self, error);
+        }
+
+        trace!("{:?} received {:?} on NodeApproval.", self, groups);
+
+        for group in &groups {
+            for pub_id in &group.1 {
+                if !self.peer_mgr.routing_table().has(pub_id.name()) {
+                    debug!("{:?} Sending connection info to {:?} on NodeApproval.",
+                           self,
+                           pub_id);
+                    let src = Authority::ManagedNode(*self.name());
+                    let node_auth = Authority::ManagedNode(*pub_id.name());
+                    if let Err(error) = self.send_connection_info_request(*pub_id, src, node_auth)
+                        .extract(&mut events) {
+                        debug!("{:?} - Failed to send connection info to {:?}: {:?}",
+                               self,
+                               pub_id,
+                               error);
+                    }
+                }
+            }
+        }
+
+        events.add_event(Event::Connected);
+        for name in self.peer_mgr.routing_table().iter() {
+            events.add_event(Event::NodeAdded(*name, self.peer_mgr.routing_table().clone()));
+        }
+        self.is_approved = true;
+
+        events.with_value(Ok(()))
+    }
+
+    fn handle_resource_proof_request(&mut self,
+                                     peer_id: PeerId,
+                                     seed: Vec<u8>,
+                                     _target_size: u32,
+                                     _difficulty: u32)
+                                     -> Evented<Result<(), RoutingError>> {
+        let direct_message = DirectMessage::ResourceProofResponse {
+            proof: seed,
+            leading_zero_bytes: 0,
+        };
+        self.send_direct_message(&peer_id, direct_message).to_evented()
```

Comment:
Does this method need to return an `Evented` at all? Couldn't this be just a `Result`?